### PR TITLE
Begin DSS builder

### LINKED ISSUE #1140: Release and deployment automation
Our release SOP is:
- Push code to master
- DSS CI tests pass (Travis does this)
- Deploy dev (from master) (Travis does this)
- Release code from master to integration (Travis has no github permissions to do this)
- Deploy integration (Travis can do this)
- DCP CI tests pass (Travis does this)
- Release code from integration to staging (Travis has no github permissions to do this)
- Deploy staging (Travis can do this)
- Release code from staging to production (this should be done manually by an operator)
- Deploy production (Travis has no AWS permissions to do this)

The steps above that Travis has no permissions to do, should be done using a deployment system running on AWS with both github and AWS permissions to perform the steps. This issue is about completing that system.

Current status: the build box is operational and can be deployed idempotently with a single command line script. It subscribes to updates in the github repo using a github webhook relaying events through an SNS-SQS bus. However the release and deploy process has not succeeded yet due to minor configuration bootstrapping/parameterization issues. The next step is to fix those issues, commit this command line script, which contains branch-specific configurations for the steps above, and also commit a serialization of the IAM policy for the build box role.

### PR DESCRIPTION
**Pull request by kislyuk**
Fixes #1140 
### Test plan
Launched the builder and verified that it successfully picks up deployment objects and executes build scripts.

Work remains to:
- Ensure that the builder has appropriate IAM permissions to run the deployment
- Replace the deploy stage in `.travis.yml` with a `make create-github-deployment` or equivalent invocation
- Hook up Slack notifications for deployments

### CODE REVIEW

File: dss-builder/rootfs.skel/usr/bin/dss-deploy-pilot [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+The agent polls GitHub for new deployment objects, which are created by running "make .
+"""
+
+import os, sys, subprocess, json, logging, argparse, shutil, signal, socket, time
+import boto3, requests
+from datetime import datetime
+
+def get_metadata(path):
+    return requests.get("http://169.254.169.254/latest/meta-data/{}".format(path)).content.decode()
```
Comment (Bento007) [inline dss-builder/rootfs.skel/usr/bin/dss-deploy-pilot]: Why is the IP address hardcoded?

Comment (kislyuk) [reply dss-builder/rootfs.skel/usr/bin/dss-deploy-pilot]: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html

**Comment by kislyuk** [comment]
@mweiden please review

File: dss-builder/rootfs.skel/usr/bin/dss-deploy-pilot [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        shutil.rmtree(d, ignore_errors=True)
+
+def run_git_command(*cmd, **kwargs):
+    kwargs["env"] = dict(kwargs.get("env", os.environ), GIT_SSH_COMMAND="aegea-git-ssh-helper")
+    return subprocess.check_call(["git"] + list(cmd), **kwargs)
+
+def clone_and_build(branch, destination):
+    if not os.path.exists(os.path.dirname(destination)):
+        os.makedirs(os.path.dirname(destination))
+    deploy_dir = "{}-{}-{}".format(args.gh_repo_name, branch, datetime.now().strftime("%Y-%m-%d-%H-%M-%S"))
```
Comment (mweiden) [inline dss-builder/rootfs.skel/usr/bin/dss-deploy-pilot]: using args as a global?

File: dss-builder/rootfs.skel/usr/bin/dss-deploy-pilot [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    parser.add_argument("--deploy-env", default=os.environ.get("GITHUB_DEPLOY_ENV"))
+    parser.add_argument("--poll-interval-seconds", default=int(os.environ.get("GITHUB_POLL_INTERVAL_SECONDS", 30)))
+    parser.add_argument("--build-location", default="/opt")
+    parser.add_argument("--deploy-key-name")
+    args = parser.parse_args()
+
+    args.github_url = "git@github.com:{}/{}".format(args.gh_owner_name, args.gh_repo_name)
+    args.build_location = os.path.abspath(os.path.join(args.build_location, args.gh_owner_name, args.gh_repo_name))
+    if not args.deploy_key_name:
+        args.deploy_key_name = "deploy.{}.{}".format(args.gh_owner_name, args.gh_repo_name)
```
Comment (mweiden) [inline dss-builder/rootfs.skel/usr/bin/dss-deploy-pilot]: This seems to be the only place where `deploy_key_name` is used?

Comment (kislyuk) [reply dss-builder/rootfs.skel/usr/bin/dss-deploy-pilot]: Deleted, thanks.

File: dss-builder/rootfs.skel/usr/bin/dss-deploy-pilot (hunk 2 above)
Comment (kislyuk) [reply dss-builder/rootfs.skel/usr/bin/dss-deploy-pilot]: Yes - something wrong with that? This is a monolithic script.

Comment (mweiden) [reply dss-builder/rootfs.skel/usr/bin/dss-deploy-pilot]: Fair enough, that's fine

**Comment by kislyuk** [comment]
Ignoring spurious ES test failure.

**Pull request merged**